index.js

```diff
@@ -3,53 +3,309 @@ const state = {
+      const storeItemImage = document.createElement(`img`);
+      storeItemImage.setAttribute(`src`, `assets/icons/${currentItem.id}.svg`);
+      storeItemImage.setAttribute(`alt`, currentItem.name);
+      const storeItemButton = document.createElement(`button`);
+      storeItemButton.innerText = `Add to cart`;
+  storeItemDiv.append(storeItemImage);
+  storeItem.append(storeItemDiv);
+  storeItem.append(storeItemButton);
+
+  return storeItem;
+}
+
+// Add click action to store items
+function storeEventListener(storeItem, i) {
+  const storeItemButton = storeItem.querySelector(`button`);
+  storeItemButton.addEventListener("click", () => {
+    updateBasket(i);
+    renderBasket()
+  });
+}
+
+// Updates cart array with new item or increased quantity
+function updateBasket(i) {
+  let currentItem = state.items[i];
+  if (state.cart.length === 0) {
+    state.cart.push({ ...currentItem, quantity: 1 });
+    return;
+  } else if (state.cart.length !== 0) {
+    for (let i = 0; i < state.cart.length; i++) {
+      if (currentItem.id === state.cart[i].id) {
+        state.cart[i].quantity++;
+        return;
+      }
+    }
+    state.cart.push({ ...currentItem, quantity: 1 });
+    return;
+  }
+}
+
+// Renders the basket area
+function renderBasket() {
+  const { basketList } = elementConstants();
+  basketList.innerHTML = ``
+  for (let i = 0; i < state.cart.length; i++) {
+    if (state.cart[i].quantity > 0) {
+    const basketItem = createBasketItem(i);
+    addButtonEventListener(basketItem, i)
+    removeButtonEventListener(basketItem, i)
+    basketList.append(basketItem)
+    }
+  }
+  updatePrice()
+}
+
+// Creates the basket items & calculates total price of each item
+function createBasketItem(i) {
+  const currentItem = state.cart[i]
+  let totalPrice = currentItem.price * currentItem.quantity
+  currentItem.total = totalPrice.toFixed(2)
+  const basketItem = document.createElement(`li`)
+  basketItem.innerHTML = `
+  <img
+    class="cart--item-icon"
+    src="assets/icons/${currentItem.id}.svg"
+    alt="${currentItem.name}"
+  />
+  <p>${currentItem.name}</p>
+  <button class="quantity-btn remove-btn center">-</button>
+  <span class="quantity-text center">${currentItem.quantity}</span>
+  <button class="quantity-btn add-btn center">+</button>
+  <h3>£${currentItem.total}</h3>
+  `
+  return basketItem
+}
+
+// Adds click response to add button
+function addButtonEventListener(basketItem, i) {
+  const addButton = basketItem.querySelector(`.add-btn`)
+  addButton.addEventListener(`click`, () => {
+    state.cart[i].quantity++
+    renderBasket()
+  })
+}
+
+// Adds click response to remove button
+function removeButtonEventListener(basketItem, i) {
+  const removeButton = basketItem.querySelector(`.remove-btn`)
+  removeButton.addEventListener(`click`, () => {
+    state.cart[i].quantity--
+    renderBasket()
+  })
+}
+
+// Updates total price
+function updatePrice() {
+  const { totalPrice } = elementConstants();
```

Comment:
So you've really taken advantage of functions and calling them in different parts of the code... The code looks really structured and easy to read...

Reply:
Agreed. This is a very good example of just breaking everything into functions.